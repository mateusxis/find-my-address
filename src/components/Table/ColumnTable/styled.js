import styled from 'styled-components';

export const SColumn = styled.td`
  flex-basis: 100%;
  display: flex;
  padding: 10px 0;
  &:before {
    color: #6c7a89;
    padding-right: 10px;
    content: attr(data-label);
    flex-basis: 50%;
    text-align: right;
  }

  @media (min-width: 961px) {
    &:before {
      content: '';
      flex-basis: 0%;
    }

    .col-1 {
      flex-basis: 40%;
    }

    .col-2 {
      flex-basis: 25%;
    }

    .col-3 {
      flex-basis: 25%;
    }

    .col-4 {
      flex-basis: 10%;
    }
  }
`;
