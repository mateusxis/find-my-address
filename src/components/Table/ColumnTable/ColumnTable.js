import { SColumn } from './styled';

const ColumnTable = ({ children, label, className }) => {
  return (
    <SColumn data-label={label} className={className}>
      {children}
    </SColumn>
  );
};

export default ColumnTable;
