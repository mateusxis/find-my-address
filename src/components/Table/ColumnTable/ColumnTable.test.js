import renderer from 'react-test-renderer';

import ColumnTable from './ColumnTable';

describe('ColumnTable', () => {
  it('should render ColumnTable', () => {
    const tree = renderer
      .create(
        <ColumnTable data-label="City" className="c1">
          New York
        </ColumnTable>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
