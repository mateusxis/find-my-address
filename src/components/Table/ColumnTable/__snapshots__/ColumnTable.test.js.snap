// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnTable should render ColumnTable 1`] = `
<td
  className="sc-bdvvtL eOJgfb c1"
>
  New York
</td>
`;
